### PR TITLE
Round code cleanup

### PR DESCRIPTION
**Pull request by ejv2**
# Pull request regarding round code cleanup

The new round system is finally done and is ready for the production game. This makes the code more manageable along with adding features and stabilising the existing game.

## What have I changed?

Each category of round is now separated into its own class. All of these rounds inherit from a single "base round". This normalises a lot of behaviour.

In addition, this fixes certain issues that the original system had. For example, behaviour involving custom intermission screens and rounds which do not ask questions (e.g skip rounds).

All of the problems which the original round system had are now much better and the code is so much more manageable.

### Files changes

1. Main.cpp
1. include/Round.h
1. include/Round.cpp
1. include/Constants.h
1. include/Scoring.h
1. include/Scoring.cpp
1. Makefile
1. .gitignore
1. Created include/round and all files inside

### Behavior changes

There are several small changes to the way rounds work under this system, but the same basic behaviour is the same. In fact, a lot of code was moved from the original round system, meaning the actual code is the same.

### Bugs/Vulnerabilities/Problems fixed

1. You can no longer get multiple skip rounds in a row
1. Division answer marking is more accurate

## Why have I changed these/Why is this necessary?

This cleanup was in desperate need for a while. Now it's here.

### Requirements

-   [x] My code includes no syntax errors
-   [x] My code is cleanly written and can be read with ease
-   [x] My code and all comments are written in the English language
-   [x] My code is formatted with Unix line endings
-   [x] I have updated the project makefile, if required
-   [x] **OPTIONAL:** My commits have adequate commit messages
-   [x] **OPTIONAL:** My commits are signed with an OpenPGP signature

### My checklist

-   [x] Reformatted round system into manageable OOP based system
-   [x] Wrote the new system to behave very similarly to the original system
-   [x] Improved and cleaned behaviour for many rounds

## Summary

In summary, this new system will make the code more manageable and easier to read.

### CODE REVIEW

**Comment by OverEngineeredCode** [comment]
[Clean up code using OOP for different round types](https://trello.com/c/Yd7X51yO/3-clean-up-code-using-oop-for-different-round-types)

**Comment by OverEngineeredCode** [comment]
A somewhat substantial amount of work now merged. This is now closed.
